### ISSUE TITLE
forEach only streams one line

### ISSUE THREAD

**Issue by hammadzz** [bhoriuchi/readline-promise#7]
I switched from `readline` to `readline-promise`

 changing from 
`rlp.on('line', (line)=> console.log(line))` to
`rlp.forEach((line)=> console.log(line))` 
now only prints a single line but does return a promise

**Comment by hammadzz** [comment]
```
import * as aws from 'aws-sdk'
import readline from 'readline-promise'

const s3 = new aws.S3({apiVersion: '2006-03-01'});
const bucket = "your-bucket"
const key="test.txt"

const reader = (bucket, key) => readline.createInterface({
  input: s3.getObject({Bucket: bucket, Key: key}).createReadStream()
})

const run = async(bucket, key) => {
const testfile = reader(bucket, key)
 await testfile.forEach((line) => console.log(line))
}

run(bucket, key)
```

**Comment by stefanberkner** [comment]
I do have the same issue. Also using a S3 Object.

**Comment by bhoriuchi** [comment]
I will look at this when i get a chance. thanks.

**Comment by bhoriuchi** [comment]
sorry for the long wait. this is fixed in v1.0.4 published to nmp. the issue was that there was no check on the remaining queue length before returning done if the done event from the read stream had already fired.
